### ISSUE TITLE
mcpr file cannot be deserialized by ReplayMod

### ISSUE THREAD

**Issue by reaganiwadha** [senseiwells/ServerReplay#3]
```
Minecraft version: 1.20.4
Fabric Loader: 0.15.3
Client ReplayMod: v2.6.14
ReplayStudio: 6cd39b0874
```

I got the ServerReplay mod running on Fabric Server, and have got it running to save the MCPR files.
```
Minecraft 1.20.4 with Fabric Loader 0.15.3
```

```
Starting net.fabricmc.loader.impl.game.minecraft.BundlerClassPathCapture
[06:38:25] [main/INFO]: Loading Minecraft 1.20.4 with Fabric Loader 0.15.3
[06:38:26] [ForkJoinPool-1-worker-2/WARN]: Mod com_github_replaymod_replaystudio uses the version 6cd39b0874 which isn't compatible with Loader's extended semantic version format (Could not parse version number component '6cd39b0874'!), SemVer is recommended for reliably evaluating dependencies and prioritizing newer version
[06:38:26] [main/INFO]: Loading 57 mods:
	- fabric-api 0.92.0+1.20.4
	   |-- fabric-api-base 0.4.35+78d798af4f
	   |-- fabric-api-lookup-api-v1 1.6.45+78d798af4f
	   |-- fabric-biome-api-v1 13.0.15+78d798af4f
	   |-- fabric-block-api-v1 1.0.14+78d798af4f
	   |-- fabric-block-view-api-v2 1.0.3+78d798af4f
	   |-- fabric-blockrenderlayer-v1 1.1.45+78d798af4f
	   |-- fabric-client-tags-api-v1 1.1.6+78d798af4f
	   |-- fabric-command-api-v1 1.2.40+f71b366f4f
	   |-- fabric-command-api-v2 2.2.19+78d798af4f
	   |-- fabric-commands-v0 0.2.57+df3654b34f
	   |-- fabric-containers-v0 0.1.81+df3654b34f
	   |-- fabric-content-registries-v0 5.0.9+78d798af4f
	   |-- fabric-convention-tags-v1 1.5.9+78d798af4f
	   |-- fabric-crash-report-info-v1 0.2.22+78d798af4f
	   |-- fabric-data-generation-api-v1 13.1.17+78d798af4f
	   |-- fabric-dimensions-v1 2.1.60+78d798af4f
	   |-- fabric-entity-events-v1 1.5.28+4ced05924f
	   |-- fabric-events-interaction-v0 0.7.0+389931eb4f
	   |-- fabric-events-lifecycle-v0 0.2.71+df3654b34f
	   |-- fabric-game-rule-api-v1 1.0.45+78d798af4f
	   |-- fabric-item-api-v1 2.1.34+78d798af4f
	   |-- fabric-item-group-api-v1 4.0.20+78d798af4f
	   |-- fabric-key-binding-api-v1 1.0.40+78d798af4f
	   |-- fabric-keybindings-v0 0.2.38+df3654b34f
	   |-- fabric-lifecycle-events-v1 2.2.29+78d798af4f
	   |-- fabric-loot-api-v2 2.1.4+78d798af4f
	   |-- fabric-message-api-v1 6.0.4+78d798af4f
	   |-- fabric-mining-level-api-v1 2.1.59+78d798af4f
	   |-- fabric-model-loading-api-v1 1.0.7+78d798af4f
	   |-- fabric-models-v0 0.4.6+9386d8a74f
	   |-- fabric-networking-api-v1 3.1.2+78d798af4f
	   |-- fabric-object-builder-api-v1 13.0.8+06274a474f
	   |-- fabric-particles-v1 1.1.6+78d798af4f
	   |-- fabric-recipe-api-v1 2.0.15+78d798af4f
	   |-- fabric-registry-sync-v0 4.0.13+78d798af4f
	   |-- fabric-renderer-api-v1 3.2.3+78d798af4f
	   |-- fabric-renderer-indigo 1.5.3+78d798af4f
	   |-- fabric-renderer-registries-v1 3.2.50+df3654b34f
	   |-- fabric-rendering-data-attachment-v1 0.3.41+73761d2e4f
	   |-- fabric-rendering-fluids-v1 3.0.32+78d798af4f
	   |-- fabric-rendering-v0 1.1.53+df3654b34f
	   |-- fabric-rendering-v1 3.0.12+78d798af4f
	   |-- fabric-resource-conditions-api-v1 2.3.13+78d798af4f
	   |-- fabric-resource-loader-v0 0.11.15+78d798af4f
	   |-- fabric-screen-api-v1 2.0.16+78d798af4f
	   |-- fabric-screen-handler-api-v1 1.3.50+78d798af4f
	   |-- fabric-sound-api-v1 1.0.16+78d798af4f
	   |-- fabric-transfer-api-v1 4.0.6+78d798af4f
	   \-- fabric-transitive-access-wideners-v1 5.0.13+78d798af4f
	- fabricloader 0.15.3
	   \-- mixinextras 0.3.2
	- java 17
	- luckperms 5.4.113
	- minecraft 1.20.4
	- serverreplay 1.0.0
	   |-- com_github_replaymod_replaystudio 6cd39b0874
	   |-- fabric-language-kotlin 1.9.2+kotlin.1.8.10
	   |    |-- org_jetbrains_kotlin_kotlin-reflect 1.8.10
	   |    |-- org_jetbrains_kotlin_kotlin-stdlib 1.8.10
	   |    |-- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.8.10
	   |    |-- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.8.10
	   |    |-- org_jetbrains_kotlinx_atomicfu-jvm 0.20.0
	   |    |-- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4
	   |    |-- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4
	   |    |-- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0
	   |    |-- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.5.0
	   |    |-- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.5.0
	   |    \-- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.5.0
	   \-- fabric-permissions-api-v0 0.3-SNAPSHOT
[06:38:26] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/Users/redacted/MCServers/NoobaFabric2/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
[06:38:26] [main/INFO]: Compatibility level set to JAVA_17
[06:38:27] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.3.2).
[06:38:40] [main/INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, name=PROD]
[06:38:43] [main/INFO]: Loaded 7 recipes
[06:38:43] [main/INFO]: Loaded 1271 advancements
[06:38:44] [main/INFO]: Applied 0 biome modifications to 0 of 64 new biomes in 2.976 ms
[06:38:44] [Server thread/INFO]:         __
[06:38:44] [Server thread/INFO]:   |    |__)   LuckPerms v5.4.113
[06:38:44] [Server thread/INFO]:   |___ |      Running on Fabric - fabric@0.92.0+1.20.4
[06:38:44] [Server thread/INFO]:
[06:38:44] [Server thread/INFO]: Loading configuration...
[06:38:44] [Server thread/INFO]: Loading storage provider... [H2]
[06:38:45] [Server thread/INFO]: Loading internal permission managers...
[06:38:45] [Server thread/INFO]: Performing initial data load...
[06:38:45] [Server thread/INFO]: Successfully enabled. (took 931ms)
[06:38:45] [Server thread/INFO]: Starting minecraft server version 1.20.4
[06:38:45] [Server thread/INFO]: Loading properties
[06:38:45] [Server thread/INFO]: Default game type: SURVIVAL
[06:38:45] [Server thread/INFO]: Generating keypair
[06:38:45] [Server thread/INFO]: Starting Minecraft server on *:25565
[06:38:45] [Server thread/INFO]: Using default channel type
[06:38:45] [Server thread/INFO]: Preparing level "world"
[06:38:48] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[06:38:52] [Worker-Main-2/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-2/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-2/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-7/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-2/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-2/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-3/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-6/INFO]: Preparing spawn area: 0%
[06:38:52] [Worker-Main-6/INFO]: Preparing spawn area: 0%
[06:38:53] [Worker-Main-3/INFO]: Preparing spawn area: 0%
[06:38:53] [Worker-Main-2/INFO]: Preparing spawn area: 16%
[06:38:54] [Worker-Main-5/INFO]: Preparing spawn area: 58%
[06:38:54] [Server thread/INFO]: Time elapsed: 6023 ms
[06:38:54] [Server thread/INFO]: Done (8.809s)! For help, type "help"
[06:38:58] [User Authenticator #1/INFO]: UUID of player PCMEncoding is d89dd8aa-b0a2-40b5-80fe-5d1faffc58e9
[06:39:01] [Server thread/INFO]: PCMEncoding[/127.0.0.1:50393] logged in with entity id 217 at (43.73729124636857, 73.01479221719906, 10.208757500076073)
[06:39:01] [Server thread/INFO]: PCMEncoding joined the game
Jan 02, 2024 6:39:18 AM com.replaymod.replaystudio.lib.viaversion.protocols.protocol1_13to1_12_2.blockconnections.ConnectionData init
INFO: Loading block connection mappings ...
Jan 02, 2024 6:39:19 AM com.replaymod.replaystudio.lib.viaversion.protocols.protocol1_13to1_12_2.storage.BlockConnectionStorage <clinit>
INFO: Using FastUtil Long2ObjectOpenHashMap for block connections
[06:39:20] [Server thread/INFO]: [PCMEncoding: Successfully started 1 recordings]
[06:39:20] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 2464ms or 49 ticks behind
[06:39:20] [Server thread/WARN]: PCMEncoding moved too quickly! -5.912492466682153,0.0,8.228081905298616
[06:39:48] [Server thread/INFO]: [PCMEncoding: Successfully stopped 1 recordings]
[06:39:49] [pool-6-thread-1/INFO]: Successfully closed replay and saved to /Users/redacted/MCServers/NoobaFabric2/./recordings/d89dd8aa-b0a2-40b5-80fe-5d1faffc58e9/2024-01-02--06-39-48.mcpr
[06:41:16] [Server thread/INFO]: PCMEncoding lost connection: Disconnected
```

However, when I load the mcpr files on to ReplayMod, it crashes. Using the mcpr files recorded by the mod works fine.

```
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: Caused by: java.io.IOException: Invalid tag id: -81
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_39878(class_4614.java:58)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_23264(class_4614.java:63)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_23262(class_4614.java:56)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2507.method_52892(class_2507.java:185)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	... 12 more
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 86 out of bounds for length 10
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 113 out of bounds for length 10
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 37 out of bounds for length 10
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:41:41] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: net.minecraft.class_8911: Loading NBT data
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2507.method_52892(class_2507.java:190)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2507.method_52894(class_2507.java:144)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2540.method_30616(class_2540.java:581)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2540.method_29171(class_2540.java:123)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2540.method_52954(class_2540.java:118)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2540.method_54148(class_2540.java:501)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2661.<init>(class_2661.java:15)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:513)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: Caused by: java.io.IOException: Invalid tag id: -80
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_39878(class_4614.java:58)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_23264(class_4614.java:63)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_4614$1.method_23262(class_4614.java:56)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2507.method_52892(class_2507.java:185)
[06:41:42] [replaymod-async-sender/INFO]: [STDERR]: 	... 12 more
```

Am I doing something wrong?

**Comment by reaganiwadha** [comment]
Related mcpr file
[2024-01-02--06-39-48.mcpr.zip](https://github.com/senseiwells/ServerReplay/files/13811109/2024-01-02--06-39-48.mcpr.zip)

**Comment by senseiwells** [comment]
I couldn't reproduce this in my dev environment.
I compiled the version I tested on [here](https://cdn.discordapp.com/attachments/967529963106492448/1192239756399104030/ServerReplay-1.0.0.jar?ex=65a85acc&is=6595e5cc&hm=14f3e76f9da6bd327394c465a5fda0a485af5d71b08dd517102cd591596f8a9f&).

**Comment by reaganiwadha** [comment]
Still the same issue
```
[06:01:54] [Render thread/INFO]: Opening replay in viewer: /Users/redacted/Library/Application Support/PrismLauncher/instances/1.20.4(1)/.minecraft/replay_recordings/2024-01-05--06-01-25.mcpr
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 41 out of bounds for length 10
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 11 out of bounds for length 10
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 54 out of bounds for length 10
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_4532.method_22310(class_2539.java:512)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at net.minecraft.class_2539$class_8698.method_52923(class_2539.java:604)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.deserializePacket(FullReplaySender.java:484)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:430)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:1123)
[06:01:54] [replaymod-async-sender/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by senseiwells** [comment]
This has now been fixed in e108063c09b6b1f513b2cbd800846a7694cced45
It turned out to be a conflict with fabric api, and I had not installed in my environment.

**Comment by reaganiwadha** [comment]
Thank you for the update!
